interview-questions.md

```diff
@@ -16,42 +16,46 @@ Researched answer:
 
 2. Which JavaScript operators will return a Boolean value?
 
 Your answer: Operators that compare things, such as <, >, <=, >= will return a Boolean value, which is true or false. 
 
-Researched answer:
+Researched answer: A Boolean value, which would be true or false, will be returned whenever you compare two things. An example of this would be in 4 > 3, which would come up as true, because four is greater than 3, and if you put 4 < 3 it wouold return false, because four is not less than 3. 
 
 3. What is an index? What is the difference between index and value?
 
 Your answer: An index is the location on a string of a character. A value is the number of characters in a string
 
-Researched answer:
+Researched answer: An index is the location of an element on a string, while a value is a type of data that is stored. For example, in the array [1, 4, 23, 15, 90], the number 23 has the index 2, since they start at 0, and the value of 23. Similarly, if you had a string, "Hello World!" the index of te letter o in Hello would be 4, and the value would be "o"
 
 4. What is iteration? 
 
 Your answer: Iteration is in a loop, whenever you have to do something multiple times to get a result. 
 
-Researched answer:
-
+Researched answer: Iteration is when a process in repeated multiple times. This comes in handy when we use loops, as the goal of those is to have many processes happen without us having to repeatedly run them over and over again. An example of this would be in the loop
+    const num = 1
+    for(let i = 0; i < 20; i++) {
+        return (num[i]) 
+    }
+The loop is repeatedly checking over and over again until it reaches 20, so that I don't have to write it 20 times to get the same result. 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
 Your answer: One person, the driver, will share their screen and do all of the coding, while listening to the navigator. The navigator is the one who tells the driver what to do, and guides them through any problems that may come up. Every 20-30 minutes the driver and navigator should push their work through github and switch roles, so that they both have a chance to fill each role. 
 
-Researched answer:
+Researched answer: There are two people, a driver and a navigator. The driver does all of the typing, the coding, and listens to what the navigator is saying. Conversly, the navigator does none of the typing, watches the drivers screen, and tells them what to write. The navigator helps the driver through all the problems they may have. Through github, they can save and share the code they are working on to each other, and every 20-30 minutes, they will do so and switch roles. 
 
 ## Looking Ahead: Terms for Next Week
 
```

Comment:
Looking ahead at the terms so you have some context when hearing the terms during lectures is a great way to prepare.